Sources/PostgresNIO/New/Connection State Machine/ExtendedQueryStateMachine.swift

```diff
@@ -258,20 +276,40 @@ struct ExtendedQueryStateMachine {
                 return .wait
             }
 
+        case .rowDescriptionReceived(let queryContext, let columns):
+            switch queryContext.query {
+            case .simpleQuery(_, let eventLoopPromise):
+                // When receiving a data row, we must ensure that the data row column count
+                // matches the previously received row description column count.
+                guard dataRow.columnCount == columns.count else {
+                    return self.setAndFireError(.unexpectedBackendMessage(.dataRow(dataRow)))
+                }
+
+                return self.avoidingStateMachineCoW { state -> Action in
+                    var demandStateMachine = RowStreamStateMachine()
+                    demandStateMachine.receivedRow(dataRow)
+                    state = .streaming(columns, demandStateMachine)
+                    let result = QueryResult(value: .rowDescription(columns), logger: queryContext.logger)
+                    return .succeedQuery(eventLoopPromise, with: result)
+                }
```

Comment:
If a `DataRow` is received and `state` is `RowDescription`, it means we have a query that does return rows.
In that case we need to succeed the query here.